src/PrestaShopBundle/Controller/Admin/TranslationsController.php

```diff
@@ -0,0 +1,251 @@
+
+        $translationLocale = str_replace('-', '_', $locale);
+
+        if (count($translationFiles) === 0) {
+            throw new \Exception('There is no translation file available');
+        }
+
+        foreach ($translationFiles as $file) {
+            $translator->addResource('xlf', $file->getPathname(), $translationLocale, $file->getBasename('.xlf'));
+        }
+
+        $catalogue = $translator->getCatalogue($translationLocale)->all();
+        $databaseCatalogue = $this->getTranslationsInDatabase($locale);
+
+        foreach ($databaseCatalogue as $domain => $messages) {
+            foreach ($messages as $translationKey => $translationValue) {
+                $catalogue[$domain][$translationKey] = $translationValue;
+            }
+        }
+
+        ksort($catalogue);
+
+        return $catalogue;
+    }
+
+    /**
+     * @param $catalogue
+     * @return array
+     */
+    protected function makeTranslationsTree(array $catalogue)
+    {
+        $translationsTree = [];
+
+        foreach ($catalogue as $domain => $messages) {
+            $adjustedDomain = $this->adjustDomainWithAcronym($domain);
+
+            $tableisedDomain = Inflector::tableize($adjustedDomain);
+            list($basename) = explode('.', $tableisedDomain);
+            $parts = array_reverse(explode('_', $basename));
+
+            $subtree = &$translationsTree;
+
+            while (count($parts) > 0) {
+                $subdomain = ucfirst(array_pop($parts));
+
+                if (!array_key_exists($subdomain, $subtree)) {
+                    $subtree[$subdomain] = array();
+                }
+                $subtree = &$subtree[$subdomain];
+            }
+
+            $subtree = $messages;
+            $subtree['__camelized_domain'] = $domain;
+        }
+
+        return $translationsTree;
+    }
+
+    /**
+     * @param $domain
+     * @return string
+     */
+    protected function adjustDomainWithAcronym($domain)
+    {
+        $adjustedDomain = $domain;
+        if (false !== strpos($domain, 'ShopPDF')) {
+            $adjustedDomain = str_replace('ShopPDF', 'ShopPdf', $domain);
+        }
+
+        return $adjustedDomain;
+    }
+
+    /**
+     * @return mixed
+     * @throws \Exception
+     */
+    protected function getLegacyToStandardLocales()
+    {
+        $translationsDirectory = $this->getResourcesDirectory();
+
+        $legacyToStandardLocalesJson = file_get_contents($translationsDirectory . '/legacy-to-standard-locales.json');
+        $legacyToStandardLocales = json_decode($legacyToStandardLocalesJson, true);
+
+        $jsonLastErrorCode = json_last_error();
+        if (JSON_ERROR_NONE !== $jsonLastErrorCode) {
+            throw new \Exception('The legacy to standard locales JSON could not be decoded', $jsonLastErrorCode);
+        }
+
+        return $legacyToStandardLocales;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getResourcesDirectory()
```

Comment:
nice helper, what do you think about moving it directly to ``FrameworkBundleAdminController`` class ?